src/rust/rust-proto-new/src/serde_impl.rs

```diff
@@ -0,0 +1,36 @@
+use bytes::{
+    Bytes,
+    BytesMut,
+};
+use prost::Message;
+
+use crate::{
+    type_url,
+    SerDe,
+    SerDeError,
+};
+
+pub trait HasAssociatedProto<T> {
```

Comment:
Maybe this naming would be more clear?

```rust
pub(crate) trait Serializable<T> {
    type Serialized: From<T> + Message + Default + TryInto<T, Error = SerDeError>;
}
```

Reply:
Hmm.. I wonder if we can also incorporate the `TypeUrl` thing in here as well like this?

```rust
pub(crate) trait Serializable<T> {
    type Serialized: From<T> + Message + Default + TryInto<T, Error = SerDeError>;

    const TYPE_URL: &'static str;
}
```